Foundation/ProcessInfo.swift

```diff
@@ -41,13 +41,28 @@ open class ProcessInfo: NSObject {
         
     }
     
-    
-    internal static var _environment: [String : String] = {
-        return Dictionary<String, String>._unconditionallyBridgeFromObjectiveC(__CFGetEnvironment()._nsObject)
-    }()
-    
     open var environment: [String : String] {
-        return ProcessInfo._environment
+        let equalSign = Int32(UInt8(ascii: "="))
+        var env: [String : String] = [:]
+        var idx = 0
+        let envp = _CFEnviron()
+
+        while let entry = envp.advanced(by: idx).pointee {
+            if let value = strchr(entry, equalSign) {
+                let keyLen = entry.distance(to: value)
+                guard keyLen > 0 else {
+                    continue
+                }
+                if let key = NSString(bytes: entry, length: keyLen, encoding: String.Encoding.utf8.rawValue) {
```

Comment:
Is it guaranteed that all platforms use a UTF8 encoding?

Reply:
@parkera Probably not, I had  looked at `__CFGetEnvironment()` and saw the decoding of UTF-8 but not the fallback to `CFStringGetSystemEncoding()`. Does the following look more portable?

```
open var environment: [String : String] {
        let equalSign = Character("=")
        var env: [String : String] = [:]
        var idx = 0
        let envp = _CFEnviron()
        let strEncoding = String.Encoding(rawValue: CFStringConvertEncodingToNSStringEncoding(CFStringGetSystemEncoding()))

        while let entry = envp.advanced(by: idx).pointee {
            if let entry = String(cString: entry, encoding: strEncoding), let i = entry.index(of: equalSign) {
                let key = String(entry.prefix(upTo: i))
                let value = String(entry.suffix(from: i).dropFirst())
                env[key] = value
            }
            idx += 1
        }
        return env
    }
```

Reply:
That looks fine too.

Reply:
```diff
-let strEncoding = String.Encoding(rawValue: CFStringConvertEncodingToNSStringEncoding(CFStringGetSystemEncoding()))
+let strEncoding = String.defaultCStringEncoding
```

* [String.defaultCStringEncoding](https://github.com/apple/swift-corelibs-foundation/blob/458977b5bc8911832e3bdacb0af9c2b3bc676e1f/Foundation/NSStringAPI.swift#L91-L93)
* [NSString.defaultCStringEncoding](https://github.com/apple/swift-corelibs-foundation/blob/458977b5bc8911832e3bdacb0af9c2b3bc676e1f/Foundation/NSString.swift#L952-L954)